toolchain/check/inst.h

```diff
@@ -97,6 +97,23 @@ auto EvalOrAddInst(Context& context, LocT loc, InstT inst)
   return EvalOrAddInst(context, SemIR::LocIdAndInst(loc, inst));
 }
 
+// Add an instruction and return its constant value by canonicalizing it but
+// without evaluating the instruction. The instruction will need to be evaluated
+// later to have any effect.
```

Comment:
In order for a call to this to be reasonable, I think we also need the caller to have already converted the instruction into a reduced form; that should be documented somehow. (Specifically, I think the constraint that we need here is that the caller knows that evaluating the instruction would only canonicalize the operands but would otherwise produce the same instruction.)